homework_2/internal/domain/services/feedProcessorService.go

```diff
@@ -31,17 +31,26 @@ func (f *FeedProcessorService) Start(ctx context.Context) error {
 	// - when updates channel is closed, exit
 	// - when exiting, close source channel
 
-	upChan := f.feed.GetUpdates()
+	upChan1 := f.feed[0].GetUpdates()
+	upChan2 := f.feed[1].GetUpdates()
```

Comment:
This won't work if you have more than 2 feeds, which is perfectly valid considering what your constructor is accepting.
An elegant way of solving this is to create a feed merger component, similar to what Jelena has done: https://github.com/jelenanemcic/code-cadets-2021/blob/main/homework_2/internal/domain/services/feedMerger.go